File: e2e/cypress-javascript/providers/ValidationProvider.js
const { AccessibilityValidator } = require('@/interactions/validators/AccessibilityValidator');
const { LanguageValidator } = require('@/interactions/validators/LanguageValidator');
const { TaskValidator } = require('@/interactions/validators/TaskValidator');
const { TasksValidator } = require('@/interactions/validators/TasksValidator');

class ValidationProvider {
  constructor() {
    this.tasks = new TasksValidator();
    this.task = new TaskValidator();
    this.language = new LanguageValidator();
    this.accessibility = new AccessibilityValidator();
  }
}

module.exports = { ValidationProvider };

File: e2e/capybara-ruby/providers/validation_provider.rb
# frozen_string_literal: true

require_relative '../interactions/validators/accessibility_validator'
require_relative '../interactions/validators/language_validator'
require_relative '../interactions/validators/task_validator'
require_relative '../interactions/validators/tasks_validator'

class ValidationProvider
  attr_reader :tasks, :task, :language, :accessibility

  def initialize
    @tasks = TasksValidator.new
    @task = TaskValidator.new
    @language = LanguageValidator.new
    @accessibility = AccessibilityValidator.new
  end
end
